tapir/core/tapir_email_base.py

```diff
@@ -106,6 +106,35 @@ def send_to_tapir_user(self, actor: User | None, recipient: TapirUser):
             tapir_user=recipient,
         )
 
+    @staticmethod
+    def include_email_body_in_log_entry():
+        """
+        Overwrite this in your subclass with 'return False', if you don't want the mail body to be in your log-entry,
+        for example if it contains sensitive data.
+        """
+        return True
+
+    def create_log_entry(self, email, actor, tapir_user, share_owner):
+        if self.include_email_body_in_log_entry():
+            EmailLogEntry().populate(
+                email_id=self.get_unique_id(),
+                email_message=email,
+                actor=actor,
+                tapir_user=tapir_user,
+                share_owner=share_owner,
+            ).save()
+        else:
+            email.body = (
+                "The content of this mail is not saved for data protection reasons"
+            )
```

Comment:
This still modifies the email object, which I think isn't expected. I would rather update `EmailLogEntry.email_content` to be optional and `EmailLogEntry().populate()` to allow email_message to be null.